From bd2c0909eafd30c93f8a54c87efbb18e69c5c63b Mon Sep 17 00:00:00 2001
From: Jamie Gibbons <jamie.gibbons@microchip.com>
Date: Fri, 13 Jun 2025 11:23:05 +0100
Subject: [PATCH 2/2] board: beagle: beaglev_fire: fix compilation warning

Pointer not initialised prior to usage. Fix uninitialised pointer
compilation warning by ensuring correct initialisation before use.

Signed-off-by: Jamie Gibbons <jamie.gibbons@microchip.com>
---
 board/beagle/beaglev_fire/beaglev_fire.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/board/beagle/beaglev_fire/beaglev_fire.c b/board/beagle/beaglev_fire/beaglev_fire.c
index a95a6972..443ff612 100644
--- a/board/beagle/beaglev_fire/beaglev_fire.c
+++ b/board/beagle/beaglev_fire/beaglev_fire.c
@@ -47,16 +47,16 @@ int board_late_init(void)
 	struct udevice *dev;
 	struct mpfs_sys_serv *sys_serv_priv;
 
-	sys_serv_priv = devm_kzalloc(dev, sizeof(*sys_serv_priv), GFP_KERNEL);
-	if (!sys_serv_priv)
-		return -ENOMEM;
-
 	ret = uclass_get_device_by_name(UCLASS_MISC, "syscontroller", &dev);
 	if (ret) {
 		debug("%s: system controller setup failed\n", __func__);
 		return ret;
 	}
 
+	sys_serv_priv = devm_kzalloc(dev, sizeof(*sys_serv_priv), GFP_KERNEL);
+	if (!sys_serv_priv)
+		return -ENOMEM;
+
 	sys_serv_priv->dev = dev;
 
 	sys_serv_priv->sys_controller = mpfs_syscontroller_get(dev);
-- 
2.39.5

